### ISSUE TITLE
Vrapper not running on Eclipse 4.21.0

### ISSUE THREAD

**Issue by mcwayliffe** [vrapper/vrapper#846]
I just moved to a new machine at work, including a clean install of the latest Eclipse (`4.21.0`). Since I can't really work without Vim bindings, the first thing I did was install Vrapper core. However, the Vrapper icon does not show up in the toolbar, the bindings do not work, and as far as I can tell there are no vrapper-related messages in the error logs. I'm happy to debug and patch, but I have zero experience with Eclipse plugins and have no idea where to begin. Any idea what could be causing this issue?

**Comment by albertdev** [comment]
Hi,
Can you go to `Help -> About Eclipse IDE -> Installation Details -> Configuration -> View Error Log` and then see if you have any error messages related to Vrapper in there?
Though a similar place to check would be to see if the Vrapper installation worked correctly. I believe the Installation Details screen also has a tab "Plugins" where you should see a couple of Vrapper ones. If they're not there then the installation somehow deemed them incompatible and never installed anything.

**Comment by keforbes** [comment]
You could also try installing Vrapper directly from the update site rather than using the Eclipse Marketplace.  I would also suggest installing the latest unstable Vrapper build since we haven't done a "stable" release in a couple years now.

Go to `Help -> Install New Software...` then click `Add...` in the top right, give it a Name like `Vrapper (unstable)` or whatever you want and `Location: http://vrapper.sourceforge.net/update-site/unstable`.  Then you can expand the `Vrapper Unstable` group, check the box for `Vrapper`, and install.  Hopefully this will give you any errors if it fails.

**Comment by mcwayliffe** [comment]
@albertdev I do see a vrapper-related message in the logs, but I suspect it has to do with trying @keforbes suggestion to uninstall `stable` and install from `unstable` (log message below). Eclipse seems to think that Vrapper did successfully install from unstable (see screenshot), but there is still no toggle icon in the toolbar. The version of vrapper I've got now is `0.75.20210730`. Next steps? 


![Screenshot 2021-12-07 085618](https://user-images.githubusercontent.com/12615871/145042092-16c1dc28-8d8d-4535-9081-dfc4bc6e7c34.png)

```
# From Eclipse error log, prior to restarting with the latest vrapper version
!ENTRY org.eclipse.equinox.p2.engine 2 0 2021-12-07 08:43:22.237
!MESSAGE 
!SUBENTRY 1 org.eclipse.equinox.p2.touchpoint.eclipse 2 0 2021-12-07 08:43:22.237
!MESSAGE Failed to find a configured bundleInfo for: [R]net.sourceforge.vrapper.core 0.74.0.
!SUBENTRY 1 org.eclipse.equinox.p2.touchpoint.eclipse 2 0 2021-12-07 08:43:22.237
!MESSAGE Failed to find a configured bundleInfo for: [R]net.sourceforge.vrapper.eclipse 0.74.0.
!SUBENTRY 1 org.eclipse.equinox.p2.touchpoint.eclipse 2 0 2021-12-07 08:43:22.237
!MESSAGE Failed to find a configured bundleInfo for: [R]net.sourceforge.vrapper.plugin.splitEditor 0.74.0.
```

**Comment by albertdev** [comment]
@mcwayliffe  I would say the next step is to verify whether vrapper is really being loaded, which should also indicate whether it is properly installed in the first place. One way to do that is by running a few commands in Eclipse's OSGi console and copy/pasting the result in here.

Open the Console view by going to `Window > Show View > Console`. When that's open, find the little "Open Console" button to get a menu, and pick "Host OSGi Console":

![image](https://user-images.githubusercontent.com/1451362/145051733-9a17f83b-cf2d-455d-a2d0-183b58989c89.png)

When that console is open, type and run the following command: `lb net.source`. I get the following result:

```
START LEVEL 6
   ID|State      |Level|Name
  606|Resolved   |    4|Vrapper Core Library (0.75.20181126)|0.75.20181126
  607|Active     |    4|Vrapper (0.75.20181126)|0.75.20181126
  608|Resolved   |    1|Vrapper - JDT enchantments (0.75.20181126)|0.75.20181126
  609|Resolved   |    4|Vrapper argtextobj Plug-in Core (0.75.20181126)|0.75.20181126
  610|Resolved   |    4|Vrapper Clang-Format Plug-in Core (0.75.20181126)|0.75.20181126
  611|Resolved   |    4|Vrapper cycle Plug-in Core (0.75.20181126)|0.75.20181126
  612|Resolved   |    4|Vrapper exchange.vim Plug-in Core (0.75.20181126)|0.75.20181126
  613|Resolved   |    4|Vrapper indenttextobj Plug-in Core (0.75.20181126)|0.75.20181126
  614|Resolved   |    4|Vrapper ipmotion Plug-in Core (0.75.20181126)|0.75.20181126
  615|Resolved   |    4|Vrapper linetextobj Plug-in Core (0.75.20181126)|0.75.20181126
  616|Resolved   |    4|Vrapper methodtextobj Plug-in Core (0.75.20181126)|0.75.20181126
  617|Resolved   |    4|Vrapper Sneak Plug-in (0.75.20181126)|0.75.20181126
  618|Active     |    4|Vrapper Split Editor Plug-in Eclipse integration (0.75.20181126)|0.75.20181126
  619|Resolved   |    4|Vrapper subwordtextobj Plug-in Core (0.75.20181126)|0.75.20181126
  620|Resolved   |    4|Vrapper Surround Plug-in Core (0.75.20181126)|0.75.20181126
  ```

Please let me know yours.

**Comment by mcwayliffe** [comment]
@albertdev My output is considerably sparser:
```
START LEVEL 6
   ID|State      |Level|Name
   49|Resolved   |    4|SourceForge LPG (1.1.0.v201004271650)|1.1.0.v201004271650
```

**Comment by albertdev** [comment]
That's... not good. That means the plugin is not even properly installed.

On my Windows 10 install of the newest Eclipse IDE everything seems to go right. What operating system do you run, and do you happen to know what JVM you use? A quick way to get all that info is by going to `Help -> About Eclipse IDE -> Installation Details -> Configuration`, then copying the first few screenfulls and pasting them here. Once you get to `System Environment Variables` you can stop. Also be wary that you might have to anonymize your username or other personal information from there.

BTW: I hope your Eclipse install isn't located in `C:\Program Files\` ? It's possible that Windows doesn't like you to install something there, so I tend to make a folder `C:\Apps` in which I then have one or more `eclipse-4.20`, `eclipse-4.21` etc. folders.

**Comment by mcwayliffe** [comment]
It looks like Eclipse **IS** installed in `C:\Program Files`, specifically: `C:\Program Files\eclipse\jee-2021-09\eclipse\eclipse.exe`.  When you say, "Windows doesn't like you to install something" in `C:\Program Files`, what do you mean exactly (I am a Linux user normally -- this is a work computer)?

 I'm using JVM version `17.0.1` on a new installation of Windows 11. Here's the output from `Installation details -> Configuration` down to system environment variables (note that `grep`ing for `net.sourceforge.vrapper` in the full output shows nothing): 

```
*** Date: Wednesday, December 8, 2021 at 8:07:56 AM Eastern Standard Time

*** Platform Details:

*** System properties:
applicationXMI=org.eclipse.ui.workbench/LegacyIDE.e4xmi
ds.delayed.keepInstances=true
ds.delayed.keepInstances.default=true
eclipse.application=org.eclipse.ui.ide.workbench
eclipse.buildId=4.21.0.I20210906-0500
eclipse.commands=-os
win32
-ws
win32
-arch
x86_64
-showsplash
C:\Users\$USER\.p2\pool\plugins\org.eclipse.epp.package.common_4.21.0.20210910-1200\splash.bmp
-launcher
C:\Program Files\eclipse\jee-2021-09\eclipse\eclipse.exe
-name
Eclipse
--launcher.library
C:\Users\$USER\.p2\pool\plugins\org.eclipse.equinox.launcher.win32.win32.x86_64_1.2.300.v20210828-0802\eclipse_11506.dll
-startup
C:\Program Files\eclipse\jee-2021-09\eclipse\\plugins/org.eclipse.equinox.launcher_1.6.300.v20210813-1054.jar
--launcher.appendVmargs
-product
org.eclipse.epp.package.jee.product
-vm
C:\Java\jdk-17.0.1\bin\server\jvm.dll
eclipse.home.location=file:/C:/Program Files/eclipse/jee-2021-09/eclipse/
eclipse.launcher=C:\Program Files\eclipse\jee-2021-09\eclipse\eclipse.exe
eclipse.launcher.name=Eclipse
eclipse.p2.data.area=file:/C:/Users/$USER/.p2/
eclipse.p2.max.threads=10
eclipse.p2.profile=C__Program Files_eclipse_jee-2021-092_eclipse
eclipse.product=org.eclipse.epp.package.jee.product
eclipse.startTime=1638968801853
eclipse.stateSaveDelayInterval=30000
eclipse.vm=C:\Java\jdk-17.0.1\bin\server\jvm.dll
eclipse.vmargs=-Dosgi.requiredJavaVersion=11
-Dosgi.instance.area.default=@user.home/eclipse-workspace
-Dsun.java.command=Eclipse
-XX:+UseG1GC
-XX:+UseStringDeduplication
--add-modules=ALL-SYSTEM
-Dosgi.requiredJavaVersion=11
-Dosgi.dataAreaRequiresExplicitInit=true
-Dorg.eclipse.swt.graphics.Resource.reportNonDisposed=true
-Xms256m
-Xmx2048m
--add-modules=ALL-SYSTEM
-Declipse.p2.max.threads=10
-Doomph.update.url=http://download.eclipse.org/oomph/updates/milestone/latest
-Doomph.redirection.index.redirection=index:/->http://git.eclipse.org/c/oomph/org.eclipse.oomph.git/plain/setups/
-Djava.class.path=C:\Program Files\eclipse\jee-2021-09\eclipse\\plugins/org.eclipse.equinox.launcher_1.6.300.v20210813-1054.jar
equinox.init.uuid=true
file.encoding=Cp1252
file.separator=\
gosh.args=--nointeractive
guice.disable.misplaced.annotation.check=true
java.class.path=C:\Program Files\eclipse\jee-2021-09\eclipse\\plugins/org.eclipse.equinox.launcher_1.6.300.v20210813-1054.jar
java.class.version=61.0
java.home=C:\Java\jdk-17.0.1
java.io.tmpdir=C:\Users\$USER\AppData\Local\Temp\
java.library.path=C:\Program Files\eclipse\jee-2021-09\eclipse;C:\WINDOWS\Sun\Java\bin;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files (x86)\PuTTY\;C:\Program Files\Git\cmd;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\170\Tools\Binn\;C:\Users\$USER\AppData\Local\Microsoft\WindowsApps;C:\Java\jdk-17.0.1\bin;;.
java.runtime.name=OpenJDK Runtime Environment
java.runtime.version=17.0.1+12-39
java.specification.name=Java Platform API Specification
java.specification.vendor=Oracle Corporation
java.specification.version=17
java.vendor=Oracle Corporation
java.vendor.url=https://java.oracle.com/
java.vendor.url.bug=https://bugreport.java.com/bugreport/
java.version=17.0.1
java.version.date=2021-10-19
java.vm.compressedOopsMode=32-bit
java.vm.info=mixed mode, sharing
java.vm.name=OpenJDK 64-Bit Server VM
java.vm.specification.name=Java Virtual Machine Specification
java.vm.specification.vendor=Oracle Corporation
java.vm.specification.version=17
java.vm.vendor=Oracle Corporation
java.vm.version=17.0.1+12-39
jdk.debug=release
line.separator=

maven.defaultProjectBuilder.disableGlobalModelCache=true
native.encoding=Cp1252
oomph.redirection.index.redirection=index:/->http://git.eclipse.org/c/oomph/org.eclipse.oomph.git/plain/setups/
oomph.update.url=http://download.eclipse.org/oomph/updates/milestone/latest
org.apache.commons.logging.Log=org.apache.commons.logging.impl.NoOpLog
org.eclipse.equinox.launcher.splash.location=C:\Users\$USER\.p2\pool\plugins\org.eclipse.epp.package.common_4.21.0.20210910-1200\splash.bmp
org.eclipse.equinox.simpleconfigurator.configUrl=file:org.eclipse.equinox.simpleconfigurator/bundles.info
org.eclipse.m2e.log.dir=C:\Javaworkspace\.metadata\.plugins\org.eclipse.m2e.logback.configuration
org.eclipse.swt.graphics.Resource.reportNonDisposed=true
org.eclipse.swt.internal.deviceZoom=100
org.eclipse.update.resolution_url=
org.osgi.framework.executionenvironment=OSGi/Minimum-1.0, OSGi/Minimum-1.1, OSGi/Minimum-1.2, JavaSE/compact1-1.8, JavaSE/compact2-1.8, JavaSE/compact3-1.8, JRE-1.1, J2SE-1.2, J2SE-1.3, J2SE-1.4, J2SE-1.5, JavaSE-1.6, JavaSE-1.7, JavaSE-1.8, JavaSE-9, JavaSE-10, JavaSE-11, JavaSE-12, JavaSE-13, JavaSE-14, JavaSE-15, JavaSE-16, JavaSE-17
org.osgi.framework.language=en
org.osgi.framework.os.name=Windows10
org.osgi.framework.os.version=10.0.0
org.osgi.framework.processor=x86-64
org.osgi.framework.storage=C:\Users\$USER\.eclipse\org.eclipse.platform_4.21.0_296365082_win32_win32_x86_64\configuration
org.osgi.framework.system.capabilities=osgi.ee; osgi.ee="OSGi/Minimum"; version:List<Version>="1.0, 1.1, 1.2", osgi.ee; osgi.ee="JRE"; version:List<Version>="1.0, 1.1", osgi.ee; osgi.ee="JavaSE"; version:List<Version>="1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 1.7, 1.8, 9.0, 10.0, 11.0, 12.0, 13.0, 14.0, 15.0, 16.0, 17.0",osgi.ee; osgi.ee="JavaSE/compact1"; version:List<Version>="1.8, 9.0, 10.0, 11.0, 12.0, 13.0, 14.0, 15.0, 16.0, 17.0",osgi.ee; osgi.ee="JavaSE/compact2"; version:List<Version>="1.8, 9.0, 10.0, 11.0, 12.0, 13.0, 14.0, 15.0, 16.0, 17.0",osgi.ee; osgi.ee="JavaSE/compact3"; version:List<Version>="1.8, 9.0, 10.0, 11.0, 12.0, 13.0, 14.0, 15.0, 16.0, 17.0"
org.osgi.framework.system.packages=com.sun.jarsigner, com.sun.java.accessibility.util, com.sun.jdi, com.sun.jdi.connect, com.sun.jdi.connect.spi, com.sun.jdi.event, com.sun.jdi.request, com.sun.management, com.sun.net.httpserver, com.sun.net.httpserver.spi, com.sun.nio.file, com.sun.nio.sctp, com.sun.security.auth, com.sun.security.auth.callback, com.sun.security.auth.login, com.sun.security.auth.module, com.sun.security.jgss, com.sun.source.doctree, com.sun.source.tree, com.sun.source.util, com.sun.tools.attach, com.sun.tools.attach.spi, com.sun.tools.javac, com.sun.tools.jconsole, java.applet, java.awt, java.awt.color, java.awt.datatransfer, java.awt.desktop, java.awt.dnd, java.awt.event, java.awt.font, java.awt.geom, java.awt.im, java.awt.im.spi, java.awt.image, java.awt.image.renderable, java.awt.print, java.beans, java.beans.beancontext, java.io, java.lang, java.lang.annotation, java.lang.constant, java.lang.instrument, java.lang.invoke, java.lang.management, java.lang.module, java.lang.ref, java.lang.reflect, java.lang.runtime, java.math, java.net, java.net.http, java.net.spi, java.nio, java.nio.channels, java.nio.channels.spi, java.nio.charset, java.nio.charset.spi, java.nio.file, java.nio.file.attribute, java.nio.file.spi, java.rmi, java.rmi.dgc, java.rmi.registry, java.rmi.server, java.security, java.security.cert, java.security.interfaces, java.security.spec, java.sql, java.text, java.text.spi, java.time, java.time.chrono, java.time.format, java.time.temporal, java.time.zone, java.util, java.util.concurrent, java.util.concurrent.atomic, java.util.concurrent.locks, java.util.function, java.util.jar, java.util.logging, java.util.prefs, java.util.random, java.util.regex, java.util.spi, java.util.stream, java.util.zip, javax.accessibility, javax.annotation.processing, javax.crypto, javax.crypto.interfaces, javax.crypto.spec, javax.imageio, javax.imageio.event, javax.imageio.metadata, javax.imageio.plugins.bmp, javax.imageio.plugins.jpeg, javax.imageio.plugins.tiff, javax.imageio.spi, javax.imageio.stream, javax.lang.model, javax.lang.model.element, javax.lang.model.type, javax.lang.model.util, javax.management, javax.management.loading, javax.management.modelmbean, javax.management.monitor, javax.management.openmbean, javax.management.relation, javax.management.remote, javax.management.remote.rmi, javax.management.timer, javax.naming, javax.naming.directory, javax.naming.event, javax.naming.ldap, javax.naming.ldap.spi, javax.naming.spi, javax.net, javax.net.ssl, javax.print, javax.print.attribute, javax.print.attribute.standard, javax.print.event, javax.rmi.ssl, javax.script, javax.security.auth, javax.security.auth.callback, javax.security.auth.kerberos, javax.security.auth.login, javax.security.auth.spi, javax.security.auth.x500, javax.security.cert, javax.security.sasl, javax.smartcardio, javax.sound.midi, javax.sound.midi.spi, javax.sound.sampled, javax.sound.sampled.spi, javax.sql, javax.sql.rowset, javax.sql.rowset.serial, javax.sql.rowset.spi, javax.swing, javax.swing.border, javax.swing.colorchooser, javax.swing.event, javax.swing.filechooser, javax.swing.plaf, javax.swing.plaf.basic, javax.swing.plaf.metal, javax.swing.plaf.multi, javax.swing.plaf.nimbus, javax.swing.plaf.synth, javax.swing.table, javax.swing.text, javax.swing.text.html, javax.swing.text.html.parser, javax.swing.text.rtf, javax.swing.tree, javax.swing.undo, javax.tools, javax.transaction.xa, javax.xml, javax.xml.catalog, javax.xml.crypto, javax.xml.crypto.dom, javax.xml.crypto.dsig, javax.xml.crypto.dsig.dom, javax.xml.crypto.dsig.keyinfo, javax.xml.crypto.dsig.spec, javax.xml.datatype, javax.xml.namespace, javax.xml.parsers, javax.xml.stream, javax.xml.stream.events, javax.xml.stream.util, javax.xml.transform, javax.xml.transform.dom, javax.xml.transform.sax, javax.xml.transform.stax, javax.xml.transform.stream, javax.xml.validation, javax.xml.xpath, jdk.dynalink, jdk.dynalink.beans, jdk.dynalink.linker, jdk.dynalink.linker.support, jdk.dynalink.support, jdk.incubator.foreign, jdk.incubator.vector, jdk.javadoc.doclet, jdk.jfr, jdk.jfr.consumer, jdk.jshell, jdk.jshell.execution, jdk.jshell.spi, jdk.jshell.tool, jdk.management.jfr, jdk.net, jdk.nio, jdk.nio.mapmode, jdk.security.jarsigner, jdk.swing.interop, netscape.javascript, org.ietf.jgss, org.w3c.dom, org.w3c.dom.bootstrap, org.w3c.dom.css, org.w3c.dom.events, org.w3c.dom.html, org.w3c.dom.ls, org.w3c.dom.ranges, org.w3c.dom.stylesheets, org.w3c.dom.traversal, org.w3c.dom.views, org.w3c.dom.xpath, org.xml.sax, org.xml.sax.ext, org.xml.sax.helpers, sun.misc, sun.reflect
org.osgi.framework.uuid=39b35a0c-1df0-467c-8ed2-71446857a14b
org.osgi.framework.vendor=Eclipse
org.osgi.framework.version=1.10.0
org.osgi.supports.framework.extension=true
org.osgi.supports.framework.fragment=true
org.osgi.supports.framework.requirebundle=true
os.arch=amd64
os.name=Windows 10
os.version=10.0
osgi.arch=x86_64
osgi.bundles=reference:file:org.eclipse.osgi.compatibility.state_1.2.500.v20210730-0750.jar,reference:file:org.eclipse.equinox.simpleconfigurator_1.4.0.v20210315-2228.jar@1:start
osgi.bundles.defaultStartLevel=4
osgi.compatibility.bootdelegation=true
osgi.compatibility.bootdelegation.default=true
osgi.configuration.area=file:/C:/Users/$USER/.eclipse/org.eclipse.platform_4.21.0_296365082_win32_win32_x86_64/configuration/
osgi.dataAreaRequiresExplicitInit=true
osgi.framework=file:/C:/Users/$USER/.p2/pool/plugins/org.eclipse.osgi_3.17.0.v20210823-1805.jar
osgi.framework.extensions=reference:file:org.eclipse.osgi.compatibility.state_1.2.500.v20210730-0750.jar
osgi.framework.shape=jar
osgi.framework.useSystemProperties=true
osgi.frameworkClassPath=., file:C:/Users/$USER/.p2/pool/plugins/org.eclipse.osgi.compatibility.state_1.2.500.v20210730-0750.jar
osgi.install.area=file:/C:/Program Files/eclipse/jee-2021-09/eclipse/
osgi.instance.area=file:/C:/Javaworkspace/
osgi.instance.area.default=file:/C:/Users/$USER/eclipse-workspace/
osgi.logfile=C:\Javaworkspace\.metadata\.log
osgi.nl=en_US
osgi.os=win32
osgi.requiredJavaVersion=11
osgi.sharedConfiguration.area=file:/c:/Program Files/eclipse/jee-2021-09/eclipse/configuration/
osgi.splashLocation=C:\Users\$USER\.p2\pool\plugins\org.eclipse.epp.package.common_4.21.0.20210910-1200\splash.bmp
osgi.splashPath=file:/C:/Users/$USER/.p2/pool/plugins/org.eclipse.platform_4.21.0.v20210906-0500
osgi.syspath=c:\Users\$USER\.p2\pool\plugins
osgi.tracefile=C:\Javaworkspace\.metadata\trace.log
osgi.ws=win32
path.separator=;
sun.arch.data.model=64
sun.boot.library.path=C:\Java\jdk-17.0.1\bin
sun.cpu.endian=little
sun.cpu.isalist=amd64
sun.io.unicode.encoding=UnicodeLittle
sun.java.command=Eclipse
sun.jnu.encoding=Cp1252
sun.management.compiler=HotSpot 64-Bit Tiered Compilers
sun.os.patch.level=
user.country=US
user.dir=C:\Program Files\eclipse\jee-2021-09\eclipse
user.home=C:\Users\$USER
user.language=en
user.name=$USER
user.script=
user.timezone=America/New_York
user.variant=
```

**Comment by albertdev** [comment]
@mcwayliffe Thans for the info.

> It looks like Eclipse IS installed in C:\Program Files, specifically: C:\Program Files\eclipse\jee-2021-09\eclipse\eclipse.exe. When you say, "Windows doesn't like you to install something" in C:\Program Files, what do you mean exactly (I am a Linux user normally -- this is a work computer)?

Sorry for being so cryptic. I didn't want to write a lengthy paragraph about Windows only to get a reply "I'm running on macOS / Linux" while I still wanted to get the warning out just in case.

What is going on is that the `C:\Program Files\` directory is protected by Windows UAC. It can't really be compared to SELinux, but seeing how it is related to security it can be a similar pain in the behind. Basically, that directory is protected against writing, and only when you start a program with "elevated" permissions (not identical  to `sudo <program>`, but close enough) can you modify something in there. Programs developed for Windows-only frequently have a dialog built-in asking you to give temporary elevated permissions to write some changes. Other solutions consist of them creating a directory in your User directory and writing all their config settings, updates, etc. in there.

Now I haven't really looked into it whether the newest Eclipse really supports being installed in `C:\Program Files\`. I'll have to check, and that takes some time. However, if you want a quick fix and you did the Eclipse installation yourself then I would recommend to install it in some other directory for now where Windows UAC will not be an issue. If an Administrator or your boss installed it then you'll need to wait until I have some time to figure it out.  At this point I myself have a habit of installing elsewhere so I didn't even notice the problem.

Just for the record: this *has been* a problem in the past where Eclipse wouldn't even run correctly when installed in `C:\Program Files\`. There are [threads on StackOverflow from people with similar problems](https://stackoverflow.com/questions/2414181/what-folder-should-i-install-eclipse-to-on-windows) and [articles by people recommending to install somewhere else](https://dzone.com/articles/eclipse-plugin-installation). It's possible that things have improved, but at this point I can't yet tell what is going wrong at your side.

**Comment by mcwayliffe** [comment]
@albertdev Thanks a bunch for the explanation! This version of Eclipse was installed by an admin, while on my previous machine I installed it myself in a user-local area (without realizing that that might have consequences). I read [the linked stackoverflow post](https://stackoverflow.com/questions/2414181/what-folder-should-i-install-eclipse-to-on-windows) and decided to try installing `vrapper` from an Eclipse running as `admin`, as the OP there did. That got me the `vrapper` icon in the toolbar and put me in VI mode for editing! Odd that a permissions issue preventing plugin installation would not fail more loudly...
Anyway, thank you so much for the help, hopefully we've found the root cause and it will be smooth sailing now.
